standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -1861,6 +1861,19 @@ struct ScheduledQueryProgressInfo{
   4: optional string errorMessage,
 }
 
+struct GetFileListRequest {
```

Comment:
This naming of this struct could be more generic (may be a better name could be GetFileMetadataRequest). Also did you consider reusing existing/extending getFileMetadata HMS API?

Reply:
I agree with the naming. The existing getFileMetadata (AFAIK) returns file metadata based on fileId. It would not be good for our use-case (replacing recursive listing during getAcidState).